### PR TITLE
KAZOO-3307 changes to notifies to sync with kamailio-42

### PR DESCRIPTION
**Pull request by lazedo**

### CODE REVIEW

File: applications/omnipresence/src/omnip_subscriptions.erl [hunk 1]
```diff
@@ -298,6 +299,12 @@ handle_info({'timeout', Ref, ?EXPIRE_MESSAGE}=_R, #state{expire_ref=Ref, ready='
 handle_info(?TABLE_READY(_Tbl), State) ->
     lager:debug("recv table_ready for ~p", [_Tbl]),
     {'noreply', State#state{ready='true'}, 'hibernate'};
+handle_info('check_sync', #state{sync_nodes=[]} = State) ->
+    gen_listener:cast('omnipresence_shared_listener', {'ready'}),
```
Comment (jamesaimonetti) [inline applications/omnipresence/src/omnip_subscriptions.erl]: This should be a function call in omnipresence_shared_listener instead. The fact that it is a gen_listener should be hidden from external modules/processes.

File: applications/omnipresence/src/omnipresence_listener.erl [hunk 2]
```diff
@@ -122,16 +128,38 @@ handle_cast('find_subscriptions_srv', #state{subs_pid=_Pid}=State) ->
...
+handle_cast('send_sync', #state{subs_pid='undefined'}=State) ->
+    {'noreply', State};
+handle_cast('send_sync', #state{queue='undefined'}=State) ->
     {'noreply', State};
-handle_cast({'gen_listener',{'is_consuming',_IsConsuming}}, State) ->
+handle_cast('send_sync', #state{consuming='false'}=State) ->
     {'noreply', State};
+handle_cast('send_sync', #state{subs_pid=Pid, queue=Queue, consuming='true', sync='false'} = State) ->
+    Payload = wh_json:from_list(
+                [{<<"Action">>, <<"Request">>}
```
Comment (jamesaimonetti) [inline applications/omnipresence/src/omnipresence_listener.erl]: I see handlers for Action = "Start" and "End" but not "Request" in omnip_subscriptions

Comment (lazedo) [inline applications/omnipresence/src/omnipresence_listener.erl]: Request is handled in kamailio

File: core/whistle-1.0.0/src/api/wapi_presence.erl [hunk 3]
```diff
@@ -500,6 +510,30 @@ publish_flush(Req, ContentType) ->
...
+-spec sync_v(api_terms()) -> boolean().
+sync_v(Prop) when is_list(Prop) ->
+    wh_api:validate(Prop, ?SYNC_HEADERS, ?SYNC_VALUES, ?SYNC_TYPES);
+sync_v(JObj) -> sync_v(wh_json:to_proplist(JObj)).
+
+publish_sync(JObj) ->
+    publish_sync(JObj, ?DEFAULT_CONTENT_TYPE).
+publish_sync(Req, ContentType) ->
+    {'ok', Payload} = wh_api:prepare_api_payload(Req, ?SYNC_VALUES, fun ?MODULE:sync/1),
+    amqp_util:presence_publish(<<"sync">>, Payload, ContentType).
```
Comment (jamesaimonetti) [inline core/whistle-1.0.0/src/api/wapi_presence.erl]: I don't see anything here to bind for this routing key in (un)bind_q/2

Comment (lazedo) [inline core/whistle-1.0.0/src/api/wapi_presence.erl]: kamailio doesn't need a bind for sync but i added it anyway.

**Pull request closed without merging**